src/components/settings/settings/EditSettingsForm.tsx

```diff
@@ -368,11 +368,14 @@ class EditSettingsForm extends Component<IProps, IState> {
 
         // If isTwoFactorAutoCatcherEnabled is enabled, but the string input is empty, set it to the default value
         if (
-          !isTwoFactorAutoCatcherEnabled &&
+          isTwoFactorAutoCatcherEnabled &&
           values.twoFactorAutoCatcherMatcher.length === 0
         ) {
           values.twoFactorAutoCatcherMatcher =
             DEFAULT_APP_SETTINGS.twoFactorAutoCatcherMatcher;
+          // empty the store if its disabled
+        } else if (!isTwoFactorAutoCatcherEnabled) {
+          values.twoFactorAutoCatcherMatcher = '';
```

Comment:
are these changes needed? when the user disables the autocatcher, the matcher still stores (in the config file) the strings. This is intended (so that a user can disable and enable the feature, without having to edit everything if enabled again)

Reply:
if its get disabled the store is filled with a string which was a security issue for @vraravam

Reply:
yes, i had requested for this to be reset/empty if the user turned it off. But, if you feel this is better for the user, then I can be convinced.

Reply:
@vraravam I think that we should allow the user to enable/disable without changing the strings. We just have to make sure we don't catch the notifications when the feature is off (which we dont, so that's ok). Nevertheless, if you feel we should make it this way, I would suggest to change it to the default settings, and not to an empty string.

@dutscher would you be able to revert this change? this way we could merge it :D

Reply:
its done and reverted to the first implementation of your code @SpecialAro 
lets merge this :)

cheers and thanks for your support!